venv/pysnes/address_computation_helper.py

```diff
@@ -0,0 +1,67 @@
+
+
+
+# some addresses are wrapped at the bank XX bank boundery (64 KB - 16 Bit addr)
+# so if you read two bytes from XXFFFF the second byte will be read
+# from XX0000 and not from X100000!
+#
+# There are mixed address modes. For example the computation of an
+# address pointer using dp register is wrapped.
+#
+# while the address itself can cross bank bounderys.
+# Direct page (dp register) and stack pointer are wrapped at the zero bank.
+# Most likely there are bugs in wrapping implementations.
+def compute_wrapped_addr(address):
+    return address & 0x00FFFF
+
+# OPCODE abs
+def abs(address, DBR):
+    return (DBR << 16) + address # no wrapping
+
+# OPCODE abs, Y (used after stack)
+def abs_y(address, DBR, Y, is8BitY):
+    if is8BitY:
+        Y = Y & 0x00FF
+    return (DBR << 16) + address + Y # no wrapping
+
+# OPCODE abs, X
+def abs_x(address, DBR, X, is8BitX):
+    if is8BitX:
+        X = X & 0x00FF
+    return (DBR << 16) + address + X # no wrapping
+
+# OPCODE dp
+def dp(address, DP):
+    wrapped_addr = compute_wrapped_addr(address + DP) # direct page wrapping
+    return (0x00 << 16) + wrapped_addr # zero bank wrapping!
```

Comment:
I understand that this is used to stress that we read from the 0 bank, but I find the addition of 0 shifted 16 times to the left to be quit confusing when first reading it. I think to know that this is zero wrapped would be enough to understand that the "DBR part" of the address is 0x00 here.